### PR TITLE
executor: fix invalid key error of fast analyze

### PR DESCRIPTION
**Pull request by lamxTyler**
### What problem does this PR solve? 

```
[18:6]invalid key - ""
github.com/pingcap/errors.AddStack
        /home/go/pkg/mod/github.com/pingcap/errors@v0.11.4/errors.go:174
github.com/pingcap/parser/terror.(*Error).GenWithStack
        /home/go/pkg/mod/github.com/pingcap/parser@v0.0.0-20190613082312-d2cf6071823d/terror/terror.go:231
github.com/pingcap/tidb/tablecodec.DecodeRowKey
        /home/go/src/github.com/pingcap/tidb/tablecodec/tablecodec.go:205
github.com/pingcap/tidb/executor.(*AnalyzeFastExec).updateCollectorSamples
        /home/go/src/github.com/pingcap/tidb/executor/analyze.go:801
github.com/pingcap/tidb/executor.(*AnalyzeFastExec).handleBatchSeekResponse
        /home/go/src/github.com/pingcap/tidb/executor/analyze.go:872
github.com/pingcap/tidb/executor.(*AnalyzeFastExec).handleSampTasks
        /home/go/src/github.com/pingcap/tidb/executor/analyze.go:968
```
### What is changed and how it works?

We should check the validity of iterator before using them.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

 - None

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: executor/analyze.go [hunk 1]
```diff
@@ -962,7 +962,9 @@ func (e *AnalyzeFastExec) handleSampTasks(bo *tikv.Backoffer, workID int, err *e
 			if *err != nil {
 				return
 			}
-			kvMap[string(iter.Key())] = iter.Value()
+			if iter.Valid() {
```
Comment (lzmhhh123) [inline executor/analyze.go]: Are there some cases cause it `false`?

Comment (lamxTyler) [reply executor/analyze.go]: When it cannot find any key.

**Pull request merged**